app/controllers/application_controller.rb

```diff
@@ -19,9 +19,11 @@ def current_page
 
   private
 
-  # NOTE: if we ever decide to bring time travel like tariff backend, it's best if we keep this method to determine the "current date"
   def actual_date
-    Date.current
+    Date.parse(params[:start_date].to_s)
+
+    rescue ArgumentError # empty as_of param means today
```

Comment:
the comment does not reflect that the param is actually not as_of, can remove.

I would check for the start days presence then fallback to today, rather than use a rescue.